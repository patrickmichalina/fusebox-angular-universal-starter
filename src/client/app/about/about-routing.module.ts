import { AboutComponent } from './about.component'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'
import { MetaGuard } from '@ngx-meta/core'

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: '',
        component: AboutComponent,
        canActivate: [MetaGuard],
        data: {
          meta: {
            title: 'i18n.about.title',
            description: 'i18n.about.description'
          },
          response: {
            cache: {
              directive: 'no-cache'
            }
          }
        }
      }
    ])
  ],
  exports: [RouterModule]
})
export class AboutRoutingModule { }
