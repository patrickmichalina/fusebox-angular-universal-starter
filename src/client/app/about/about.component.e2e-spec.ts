import { browser } from 'protractor';

describe('About', () => {

  beforeEach(async () => {
    return await browser.get('/about');
  });

  it('should have correct feature heading', () => {
    expect(browser.getTitle()).toEqual('About - Fusebox Angular Universal Starter');
  });

});