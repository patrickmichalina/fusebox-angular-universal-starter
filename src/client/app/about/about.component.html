<h1>About</h1>
<p>An Angular starter with all the tools necessary to create server rendered Angular applications.</p>

<form [formGroup]="form" novalidate>
  <mat-form-field>
    <input matInput placeholder="Title" formControlName="title">
  </mat-form-field>
  <mat-form-field>
    <textarea matInput placeholder="html" formControlName="html"></textarea>
  </mat-form-field>
  <button mat-raised-button color="primary" (click)="create()" [disabled]="!form.valid">
    Post
  </button>
  <button mat-raised-button color="warn" (click)="removeAll()">
    Delete All Posts
  </button>
</form>

<mat-card *ngFor="let post of posts$ | async; trackBy: trackByPost">
  <h4>{{ post.title }}</h4>
  <div [innerHTML]="post.html"></div>
</mat-card>