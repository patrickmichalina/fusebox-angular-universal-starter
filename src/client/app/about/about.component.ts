import { PlatformService } from './../shared/services/platform.service'
import { ChangeDetectionStrategy, Component } from '@angular/core'
import { FirebaseDatabaseService } from '../shared/services/firebase-database.service'
import { FormControl, FormGroup, Validators } from '@angular/forms'

@Component({
  selector: 'pm-about',
  templateUrl: './about.component.html',
  styleUrls: ['./about.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AboutComponent {
  readonly posts$ = this.db.getList<{ readonly title: string, readonly html: string }>('posts').map(a => {
    return a.map((b: any) => {
      return {
        title: b.title || '',
        html: b.html || ''
      }
    })
  })

  public readonly form = new FormGroup({
    title: new FormControl('', [
      Validators.required
    ]),
    html: new FormControl('', [
      Validators.required
    ])
  })

  constructor(private db: FirebaseDatabaseService, ps: PlatformService) { }

  create() {
    this.db.getListRef('posts').push(this.form.value)
  }

  removeAll() {
    this.db.getObjectRef('posts').remove()
  }

  trackByPost(index: number, item: any) {
    return index
  }
}
