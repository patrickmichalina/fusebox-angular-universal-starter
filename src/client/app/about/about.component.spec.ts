import { AboutComponent } from './about.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { AboutModule } from './about.module'
import { AppTestingModule } from '../../../testing/app-testing.module'

@Component({
  selector: 'test-component',
  template: '<pm-about></pm-about>'
})
class TestComponent {}

describe(AboutComponent.name, () => {
  let fixture: ComponentFixture<TestComponent>

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [AppTestingModule.forRoot(), AboutModule],
      declarations: [TestComponent]
    }).compileComponents()
  }))

  beforeEach(async(() => {
    fixture = TestBed.createComponent(TestComponent)
  }))

  afterEach(async(() => {
    TestBed.resetTestingModule()
  }))

  it('should compile', async(() => {
    fixture.detectChanges()
    expect(fixture.nativeElement).toBeDefined()
    expect(fixture).toMatchSnapshot()
  }))
})
