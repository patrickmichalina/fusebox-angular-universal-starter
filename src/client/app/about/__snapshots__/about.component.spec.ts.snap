// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutComponent should compile 1`] = `
<test-component>
  <pm-about>
    <h1>
      About
    </h1>
    <p>
      An Angular starter with all the tools necessary to create server rendered Angular applications.
    </p>
    <form
      class="ng-untouched ng-pristine ng-invalid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <mat-form-field
        class="mat-input-container mat-form-field ng-tns-c0-0 mat-form-field-type-mat-input mat-form-field-can-float mat-form-field-hide-placeholder mat-primary ng-untouched ng-pristine ng-invalid"
      >
        <div
          class="mat-input-wrapper mat-form-field-wrapper"
        >
          <div
            class="mat-input-flex mat-form-field-flex"
          >
            
            <div
              class="mat-input-infix mat-form-field-infix"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-invalid"
                formcontrolname="title"
                id="mat-input-0"
                matinput=""
                ng-reflect-name="title"
                ng-reflect-placeholder="Title"
                placeholder="Title"
              />
              <span
                class="mat-form-field-label-wrapper mat-input-placeholder-wrapper mat-form-field-placeholder-wrapper"
              >
                
                <label
                  aria-owns="mat-input-0"
                  class="mat-form-field-label mat-input-placeholder mat-form-field-placeholder ng-tns-c0-0 mat-empty mat-form-field-empty ng-star-inserted"
                  for="mat-input-0"
                  ng-reflect-ng-switch="false"
                >
                  
                  
                  Title
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          <div
            class="mat-input-underline mat-form-field-underline"
          >
            <span
              class="mat-input-ripple mat-form-field-ripple"
            />
          </div>
          <div
            class="mat-input-subscript-wrapper mat-form-field-subscript-wrapper"
            ng-reflect-ng-switch="hint"
          >
            
            
            <div
              class="mat-input-hint-wrapper mat-form-field-hint-wrapper ng-tns-c0-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
              style="opacity: 1;"
            >
              
              <div
                class="mat-input-hint-spacer mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      <mat-form-field
        class="mat-input-container mat-form-field ng-tns-c0-1 mat-form-field-type-mat-input mat-form-field-can-float mat-form-field-hide-placeholder mat-primary ng-untouched ng-pristine ng-invalid"
      >
        <div
          class="mat-input-wrapper mat-form-field-wrapper"
        >
          <div
            class="mat-input-flex mat-form-field-flex"
          >
            
            <div
              class="mat-input-infix mat-form-field-infix"
            >
              <textarea
                aria-invalid="false"
                aria-required="false"
                class="mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-invalid"
                formcontrolname="html"
                id="mat-input-1"
                matinput=""
                ng-reflect-name="html"
                ng-reflect-placeholder="html"
                placeholder="html"
              />
              <span
                class="mat-form-field-label-wrapper mat-input-placeholder-wrapper mat-form-field-placeholder-wrapper"
              >
                
                <label
                  aria-owns="mat-input-1"
                  class="mat-form-field-label mat-input-placeholder mat-form-field-placeholder ng-tns-c0-1 mat-empty mat-form-field-empty ng-star-inserted"
                  for="mat-input-1"
                  ng-reflect-ng-switch="false"
                >
                  
                  
                  html
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          <div
            class="mat-input-underline mat-form-field-underline"
          >
            <span
              class="mat-input-ripple mat-form-field-ripple"
            />
          </div>
          <div
            class="mat-input-subscript-wrapper mat-form-field-subscript-wrapper"
            ng-reflect-ng-switch="hint"
          >
            
            
            <div
              class="mat-input-hint-wrapper mat-form-field-hint-wrapper ng-tns-c0-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
              style="opacity: 1;"
            >
              
              <div
                class="mat-input-hint-spacer mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      <button
        class="mat-raised-button mat-primary"
        color="primary"
        disabled=""
        mat-raised-button=""
        ng-reflect-color="primary"
        ng-reflect-disabled="true"
      >
        <span
          class="mat-button-wrapper"
        >
           Post 
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="true"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
      <button
        class="mat-raised-button mat-warn"
        color="warn"
        mat-raised-button=""
        ng-reflect-color="warn"
      >
        <span
          class="mat-button-wrapper"
        >
           Delete All Posts 
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
    </form>
    
  </pm-about>
</test-component>
`;
